### PR TITLE
Add new CarrierLogo component

### LINKED ISSUE #114: Create <CarrierLogo /> component
## Description
The main purpose of carrier logo component is to display a logo for airlines, train services, etc.

## Visual style
carrier logos

## Parameters
Parameter | Values | Default
-- | -- | --
Size | Small, Medium, Large | Medium
Carriers | ? | ?

It's possible, that we don't have logos for some carriers. In these cases, use default image instead.
**Default image:**
default image

### For all configurations
```
border-radius: $borderRadiusNormal
background-color: $backgroundCarrierLogo
```
### Additional information

- It's possible to have from 1 to 4 carriers in that container. This should be possible only for Large size. Size of the icons inside is 16x16px, with exception of 3-4 carriers (in this case, it's 15x15 as we want 2px spacing between logos).
- We upload the logo of a carrier in 128x128px size and with `png` extension. All logos are then generated in 4 different sizes (16x16, 32x32, 64x64, 128x128). For medium size icons please use 32x32 version as the original one.
- The component needs to be prepared for retina displays. 
- Carrier name should be used for `alt` and `title` of the image; even in cases when we use the default image for their logo.

 
### Based on the selected configuration

Parameter:Value | property: $token
-- | --
Size:Small | `height: $heightIconSmall`<br>`width: $widthIconSmall`
Size:Medium | `height: $heightIconMedium`<br>`width: $widthIconMedium`
Size:Large | `height: $heightIconLarge`<br>`width: $widthIconLarge`

## Our original component for carrier logo from MMB
[Check Storybook](https://www.kiwi.com/storybook/?knob-Visa%20required%3A=false&knob-Cuba%20regulation%3A=false&selectedKind=MMB%2FCarrierLogo&selectedStory=default&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs)

### PR DESCRIPTION
**Pull request by wEprous**
This Pull Request meets the following criteria:

* [x] Tests have been added/adjusted for my new feature
* [x] New Components are registered in index.js of my project

Add new CarrierLogo component.

closes #114

### CODE REVIEW

File: src/CarrierLogo/index.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  return (
+    <div>
+      {carriers
+        .slice(0, 4)
+        .map(carrierImage => (
+          <img
+            key={carrierImage.code}
+            src={`${baseURL}/airlines/${sourceSize}/${carrierImage.code}.png`}
```
Comment (jaroslav-kubicek) [inline src/CarrierLogo/index.js]: I would suggest enabling accepting the whole URL as well => `{ code: string, name: string }` => `{ code: string, name: string, url?: string }` 

and then something like ```url || `${baseURL}/airlines/${sourceSize}/${carrierImage.code}.png` ```

File: src/CarrierLogo/index.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+const retinaSizes = {
+  small: 32,
+  medium: 64,
+  large: 64,
+};
+
+type Props = {
+  size: "small" | "medium" | "large",
+  carriers: { code: string, name: string }[],
```
Comment (jaroslav-kubicek) [inline src/CarrierLogo/index.js]: I would prefer this, it's more readable:

```
type Carrier = { code: string, name: string };

type Props = {
  size: "small" | "medium" | "large",
  carriers: Carrier[],
}
```

File: src/CarrierLogo/index.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+const directorySizes = {
+  small: 16,
+  medium: 32,
+  large: 32,
+};
+
+const renderSizes = {
+  small: parseInt(tokens.heightIconSmall, 10),
+  medium: parseInt(tokens.heightIconMedium, 10),
+  large: 32, // replace it for token heightIconLarge
```
Comment (jaroslav-kubicek) [inline src/CarrierLogo/index.js]: forgotten TODO?

File: src/CarrierLogo/index.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            alt={carrierImage.name}
+            title={carrierImage.name}
+          />
+        ))}
+      <style jsx>{`
+        div {
+          border-radius: ${tokens.borderRadiusNormal};
+          background-color: #fff; // replace it for tokens.backgroundCarrierLogo
+          height: ${carriers.length > 1
+            ? "32px" // replace 32px for token heightIconLarge
```
Comment (jaroslav-kubicek) [inline src/CarrierLogo/index.js]: fotgotten TODO?

File: src/CarrierLogo/index.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            src={`${baseURL}/airlines/${sourceSize}/${carrierImage.code}.png`}
+            srcSet={`${baseURL}/airlines/${srcSetSize}/${carrierImage.code}.png 2x`}
+            alt={carrierImage.name}
+            title={carrierImage.name}
+          />
+        ))}
+      <style jsx>{`
+        div {
+          border-radius: ${tokens.borderRadiusNormal};
+          background-color: #fff; // replace it for tokens.backgroundCarrierLogo
```
Comment (jaroslav-kubicek) [inline src/CarrierLogo/index.js]: forgotten TODO?

File: src/CarrierLogo/index.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            : `${renderSizes[size]}px`};
+          display: flex;
+          flex-direction: row;
+          flex-wrap: wrap;
+          position: relative;
+        }
+        img {
+          height: ${imageSize}px;
+          width: ${imageSize}px;
+          border-radius: 3px; // {tokens.borderRadiusNormal};
```
Comment (jaroslav-kubicek) [inline src/CarrierLogo/index.js]: TODO?

File: src/CarrierLogo/index.js (hunk 2 above)
Comment (wEprous) [reply src/CarrierLogo/index.js]: OK

File: src/CarrierLogo/index.js (hunk 5 above)
Comment (wEprous) [reply src/CarrierLogo/index.js]: Will fix it.

File: src/CarrierLogo/index.js (hunk 4 above)
Comment (wEprous) [reply src/CarrierLogo/index.js]: Same as above.

File: src/CarrierLogo/index.js (hunk 6 above)
Comment (wEprous) [reply src/CarrierLogo/index.js]: Same as above.

File: src/CarrierLogo/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (wEprous) [reply src/CarrierLogo/index.js]: Will fix it.

File: src/CarrierLogo/index.js (hunk 1 above)
Comment (jaroslav-kubicek) [reply src/CarrierLogo/index.js]: ok, I see your explanation

File: src/CarrierLogo/index.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            alt={carrierImage.name}
+            title={carrierImage.name}
+          />
+        ))}
+      <style jsx>{`
+        div {
+          border-radius: ${tokens.borderRadiusNormal};
+          background-color: ${tokens.backgroundCarrierLogo};
+          height: ${carriers.length > 1 ? `${tokens.heightIconLarge}px` : `${renderSizes[size]}px`};
+          width: ${carriers.length > 1 ? `${tokens.heightIconLarge}px` : `${renderSizes[size]}px`};
```
Comment (jaroslav-kubicek) [inline src/CarrierLogo/index.js]: `heightIconLarge` => `widthIconLarge`?

**Pull request merged**